libcontainer/configs/validate/validator.go

```diff
@@ -96,7 +96,12 @@ func security(config *configs.Config) error {
 }
 
 func namespaces(config *configs.Config) error {
-	if config.Namespaces.Contains(configs.NEWUSER) {
+	// ima namespace is a part of user namespace but still is a separate namespace
+	if config.Namespaces.Contains(configs.NEWIMA) && !config.Namespaces.Contains(configs.NEWUSER) {
+		return errors.New("IMA namespace cannot be created without USER namespace")
+	}
+
+	if config.Namespaces.Contains(configs.NEWUSER) || config.Namespaces.Contains(configs.NEWIMA) {
```

Comment:
The check for NEWIMA here is redundant, as we error out above if NEWIMA is set and NEWUSER is not.

Also note that `Contains` is kinda expensive since it does a linear search, so try to optimize the code to not call it 4 times here.

Reply:
Indeed, I think it is possible to move it to the `else` statement for `NEWUSER` check like this:
```
if NEWUSER {
    // ...
} else {
    if NEWIMA {
        return errors.New(...)
    }
}
```
This way `Contains` will be called only one more time if `NEWIMA` is specified and `NEWUSER` isn't. I guess this will be sufficient.

Thanks!